volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,228 @@
+            ),
+            requirements.BooleanRequirement(
+                name="permitted",
+                description="Show only permitted capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="effective",
+                description="Show only effective capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="bounding",
+                description="Show only bounding capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="ambient",
+                description="Show only ambient capabilities in human-readable strings.",
+                optional=True,
+            ),
+        ]
+
+    def _check_capabilities_support(self):
+        """Checks that the framework supports at least as much capabilities as
+        the kernel being analysed. Otherwise, it shows a warning for the
+        developers.
+        """
+        vmlinux = self.context.modules[self.config["kernel"]]
+
+        kernel_cap_last_cap = vmlinux.object(
+            object_type="int", offset=kernel_cap_last_cap
+        )
+        vol2_last_cap = extensions.kernel_cap_struct.get_last_cap_value()
+        if kernel_cap_last_cap > vol2_last_cap:
+            vollog.warning(
+                "Developers: The supported Linux capabilities of this plugin are outdated for this kernel"
+            )
+
+    @staticmethod
+    def _decode_cap(cap: interfaces.objects.ObjectInterface) -> str:
+        """Returns a textual representation of the capability set.
+        The format is a comma-separated list of capabilitites. In order to
+        summarize the output and if all the capabilities are enabled, instead of
+        the individual capabilities, the special name "all" will be shown.
+
+        Args:
+            cap: Kernel capability object. Usually a 'kernel_cap_struct' struct
+
+        Returns:
+            str: A string with a comma separated list of decoded capabilities
+        """
+        if isinstance(cap, renderers.NotAvailableValue):
+            return cap
+
+        cap_value = cap.get_capabilities()
+        if cap_value == 0:
+            return "-"
+
+        CAP_FULL = 0xFFFFFFFF
+        if cap_value == CAP_FULL:
+            return "all"
+
+        return ", ".join(cap.enumerate_capabilities())
+
+    @classmethod
+    def get_task_capabilities(cls, task: interfaces.objects.ObjectInterface) -> Dict:
+        """Returns a dict with the task basic information along with its capabilities
+
+        Args:
+            task: A task object from where to get the fields.
+
+        Returns:
+            dict: A dict with the task basic information along with its capabilities
+        """
+        task_cred = task.real_cred
+        fields = {
```

Comment:
Hmmm, this feels like quite a specific subdivision and I can't tell if it makes the most sense?  I'm wondering if actually a named_tuple would be better here?  That way new ones can be added in and potentially change the other without distrupting other stuff (as long as everything's accessed using names)?

Other options would be fine grained names, or a set of index constants (ugly) just something so that if the data returned is extended or shifts in some way in the future, we can easily expand/adapt to it...